### ISSUE TITLE
[Feature Request] Support system available lammps

### ISSUE THREAD

**Issue by ccoulombe** [deepmodeling/deepmd-kit#4376]
### Summary

On HPC clusters, highly optimized softwares are most often already available. Allow deepmd-kit to build and link to an already available and correctly configured lammps.

### Detailed Description

Software is already available on numerous systems, allow one to build deepmd-kit with already optimized available lammps and not try to download or build it from sources.

If one defines `LAMMPS_ROOT`, for example, then prefer it to building from sources.

**Comment by njzjz** [comment]
There is the CMake option [`LAMMPS_SOURCE_ROOT`](https://docs.deepmodeling.com/projects/deepmd/en/v3.0.0rc0/install/install-from-source.html#variable:LAMMPS_SOURCE_ROOT) to build a LAMMPS plugin.

For details of the plugin mode, refer to LAMMPS documentation [Compiling plugins](https://docs.lammps.org/Developer_plugins.html#compiling-plugins) and [plugin command](https://docs.lammps.org/plugin.html).

**Comment by ccoulombe** [comment]
Yes, I did try to use it, but it needs a source tree, rather then an already built and available lammps installation.

For example, this https://github.com/deepmodeling/deepmd-kit/blob/84ca63ca0463c2c91a05bc8a5789a933edb19132/source/lmp/plugin/CMakeLists.txt#L52 says that the `src` needs to exists, but on an installed lammps, it does not, nor contains the source files.

Our system and optimized lammps is already built with the correct settings. 

That's is why it would be great if one could link on an already available installation rather then to compile one.

**Comment by njzjz** [comment]
Unlike other libraries, LAMMPS doesn't install header files required by the plugin into the system.

**Comment by ccoulombe** [comment]
What would be those files? (If you know)
Our installation do include the headers.

Thanks

**Comment by njzjz** [comment]
https://github.com/deepmodeling/deepmd-kit/blob/db2bc9447fd2c0c4c9d003e4adde778d53d39ba4/source/lmp/plugin/deepmdplugin.cpp#L8
https://github.com/deepmodeling/deepmd-kit/blob/db2bc9447fd2c0c4c9d003e4adde778d53d39ba4/source/lmp/pair_deepmd.cpp#L11-L25
https://github.com/deepmodeling/deepmd-kit/blob/db2bc9447fd2c0c4c9d003e4adde778d53d39ba4/source/lmp/pppm_dplr.cpp#L6-L17
https://github.com/deepmodeling/deepmd-kit/blob/db2bc9447fd2c0c4c9d003e4adde778d53d39ba4/source/lmp/compute_deeptensor_atom.cpp#L7-L19

**Comment by njzjz** [comment]
I will close the issue as it isn't possible unless LAMMPS adds these headers to the installation.
